tidy3d/plugins/dispersion/fit.py

```diff
@@ -25,20 +25,20 @@ class DispersionFitter(Tidy3dBaseModel):
     """Tool for fitting refractive index data to get a
     dispersive medium described by :class:`.PoleResidue` model."""
 
-    wvl_um: Union[Tuple[float, ...], ArrayFloat1D] = Field(
+    wvl_um: ArrayFloat1D = Field(
         ...,
         title="Wavelength data",
         description="Wavelength data in micrometers.",
         units=MICROMETER,
     )
 
-    n_data: Union[Tuple[float, ...], ArrayFloat1D] = Field(
+    n_data: ArrayFloat1D = Field(
         ...,
         title="Index of refraction data",
         description="Real part of the complex index of refraction.",
     )
 
-    k_data: Union[Tuple[float, ...], ArrayFloat1D] = Field(
+    k_data: ArrayFloat1D = Field(
```

Comment:
In principle, it should be `Optional[ArrayFloat1D]`?

Reply:
Good question. My understanding is that we can default to None to indicate that the value can be missing, but, if the user decides to provide it, it must be an array to pass validation. This is how it is defined in many other places, that's why I left it like this. @tylerflex, is there any reason to prefer one way or the other?

Reply:
I think technically `Optional[ArrayFloat1D]` is correct, but my understanding is that pydantic implicitly knows it's optional if the default value is `None`. So we in fact have many parts of the code that do something like this. I think in this particular case, it's fine either way. and I'd say `Optional[]` should mainly be used in method calls, or when the default value is not None (but None would be accepted). If I were writing this, I'd probably just leave it without Optional just to keep the API reference slightly more readable.

Reply:
I'll leave it as is, then.